soc/silabs/silabs_siwx91x/siwg917/nwp.c

```diff
@@ -0,0 +1,206 @@
+			SL_SI91X_BLE_MAX_NBR_CENTRALS(RSI_BLE_MAX_NBR_CENTRALS) |
+			SL_SI91X_BLE_MAX_NBR_ATT_SERV(RSI_BLE_MAX_NBR_ATT_SERV) |
+			SL_SI91X_BLE_MAX_NBR_ATT_REC(RSI_BLE_MAX_NBR_ATT_REC) |
+			SL_SI91X_BLE_PWR_INX(RSI_BLE_PWR_INX) |
+			SL_SI91X_BLE_PWR_SAVE_OPTIONS(RSI_BLE_PWR_SAVE_OPTIONS) |
+			SL_SI91X_916_BLE_COMPATIBLE_FEAT_ENABLE |
+			SL_SI91X_FEAT_BLE_CUSTOM_FEAT_EXTENSION_VALID;
+		boot_config->ble_ext_feature_bit_map |=
+			SL_SI91X_BLE_NUM_CONN_EVENTS(RSI_BLE_NUM_CONN_EVENTS) |
+			SL_SI91X_BLE_NUM_REC_BYTES(RSI_BLE_NUM_REC_BYTES) |
+			SL_SI91X_BLE_ENABLE_ADV_EXTN |
+			SL_SI91X_BLE_AE_MAX_ADV_SETS(RSI_BLE_AE_MAX_ADV_SETS);
+#endif
+	} else if (wifi_oper_mode == SL_SI91X_ACCESS_POINT_MODE) {
+		boot_config->oper_mode = SL_SI91X_ACCESS_POINT_MODE;
+		boot_config->coex_mode = SL_SI91X_WLAN_ONLY_MODE;
+	} else {
+		return -EINVAL;
+	}
+
+#ifdef CONFIG_WIFI_SILABS_SIWX91X
+	if (IS_ENABLED(CONFIG_WIFI_SILABS_SIWX91X_NET_STACK_OFFLOAD)) {
+		boot_config->tcp_ip_feature_bit_map |= SL_SI91X_TCP_IP_FEAT_ICMP;
+		boot_config->ext_tcp_ip_feature_bit_map |= SL_SI91X_EXT_TCP_IP_WINDOW_SCALING;
+		boot_config->ext_tcp_ip_feature_bit_map |= SL_SI91X_EXT_TCP_IP_TOTAL_SELECTS(10);
+
+		if (IS_ENABLED(CONFIG_NET_IPV6)) {
+			boot_config->tcp_ip_feature_bit_map |= SL_SI91X_TCP_IP_FEAT_IPV6;
+			if (wifi_oper_mode == SL_SI91X_CLIENT_MODE) {
+				boot_config->tcp_ip_feature_bit_map |=
+					SL_SI91X_TCP_IP_FEAT_DHCPV6_CLIENT;
+			} else if (wifi_oper_mode == SL_SI91X_ACCESS_POINT_MODE) {
+				boot_config->tcp_ip_feature_bit_map |=
+					SL_SI91X_TCP_IP_FEAT_DHCPV6_SERVER;
+			}
+		}
+
+		if (IS_ENABLED(CONFIG_NET_IPV4)) {
+			if (wifi_oper_mode == SL_SI91X_CLIENT_MODE) {
+				boot_config->tcp_ip_feature_bit_map |=
+					SL_SI91X_TCP_IP_FEAT_DHCPV4_CLIENT;
+			} else if (wifi_oper_mode == SL_SI91X_ACCESS_POINT_MODE) {
+				boot_config->tcp_ip_feature_bit_map |=
+					SL_SI91X_TCP_IP_FEAT_DHCPV4_SERVER;
+			}
+		}
+	} else {
+		boot_config->tcp_ip_feature_bit_map |= SL_SI91X_TCP_IP_FEAT_BYPASS;
+	}
+#endif
+	memcpy(get_config, &default_config, sizeof(default_config));
+	return 0;
+}
+
+int siwx91x_nwp_mode_switch(uint8_t oper_mode)
+{
+	sl_wifi_device_configuration_t nwp_config;
+	sl_wifi_interface_t interface;
+	sl_status_t status;
+
+	interface = sl_wifi_get_default_interface();
+	switch (oper_mode) {
+	case WIFI_STA_MODE:
+		if (FIELD_GET(SIWX91X_INTERFACE_MASK, interface) == SL_WIFI_CLIENT_INTERFACE) {
+			return 0;
+		}
+
+		siwg91x_get_nwp_config(SL_SI91X_CLIENT_MODE, &nwp_config);
+		break;
+	case WIFI_AP_MODE:
+		if (FIELD_GET(SIWX91X_INTERFACE_MASK, interface) == SL_WIFI_AP_INTERFACE) {
+			return 0;
+		}
+
+		siwg91x_get_nwp_config(SL_SI91X_ACCESS_POINT_MODE, &nwp_config);
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	/* FIXME: Calling sl_wifi_deinit() impacts Bluetooth if coexistence is enabled */
+	if (IS_ENABLED(CONFIG_BT_SILABS_SIWX91X) && oper_mode == WIFI_AP_MODE) {
+		LOG_WRN("Bluetooth is not supported in AP mode");
+	}
```

Comment:
I believe it make more sense to make this test in `siwg91x_get_nwp_config()` / `if (wifi_oper_mode == SL_SI91X_ACCESS_POINT_MODE)`.